### ISSUE TITLE
Install fails due to different issues

### ISSUE THREAD

**Issue by blwtxc** [DCS-Easyware/gsit#48]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

9.5.12

### Bug description

* `.lock` files are still present
* Some patching error
```bash
Failed to set PHP CodeSniffer installed_paths Config
> @php -r "file_put_contents('.composer.hash', sha1_file('composer.lock'));"
> patch -f -p1 -d vendor/laminas/laminas-mail/ < tools/laminas-mail-128.patch || echo 'Error applying patch, retrieving some mail attachement could fail'
patching file src/Header/ContentDisposition.php
Hunk #1 FAILED at 137.
1 out of 1 hunk FAILED -- saving rejects to file src/Header/ContentDisposition.php.rej
patching file src/Header/ContentDisposition.php
Hunk #1 succeeded at 98 (offset 11 lines).
Error applying patch, retrieving some mail attachement could fail
> patch -f -p1 -d vendor/blueimp/jquery-file-upload/ < tools/jquery-file-upload.patch || echo 'Error applying patch, will have problem to upload files on PHP 8.1'
patching file server/php/UploadHandler.php
```
* Some dependency error
```bash
[webpack-cli] Failed to load '/var/www/gsit/webpack.config.js' config
[webpack-cli] Error: Cannot find module 'ajv/dist/compile/codegen'
Require stack:
- /var/www/gsit/node_modules/ajv-keywords/dist/definitions/typeof.js
- /var/www/gsit/node_modules/ajv-keywords/dist/keywords/typeof.js
- /var/www/gsit/node_modules/ajv-keywords/dist/keywords/index.js
- /var/www/gsit/node_modules/ajv-keywords/dist/index.js
- /var/www/gsit/node_modules/schema-utils/dist/validate.js
- /var/www/gsit/node_modules/schema-utils/dist/index.js
- /var/www/gsit/node_modules/copy-webpack-plugin/dist/index.js
- /var/www/gsit/webpack.config.js
- /var/www/gsit/node_modules/webpack-cli/lib/webpack-cli.js
- /var/www/gsit/node_modules/webpack-cli/lib/bootstrap.js
- /var/www/gsit/node_modules/webpack-cli/bin/cli.js
- /var/www/gsit/node_modules/webpack/bin/webpack.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/var/www/gsit/node_modules/ajv-keywords/dist/definitions/typeof.js:3:19)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/www/gsit/node_modules/ajv-keywords/dist/definitions/typeof.js',
    '/var/www/gsit/node_modules/ajv-keywords/dist/keywords/typeof.js',
    '/var/www/gsit/node_modules/ajv-keywords/dist/keywords/index.js',
    '/var/www/gsit/node_modules/ajv-keywords/dist/index.js',
    '/var/www/gsit/node_modules/schema-utils/dist/validate.js',
    '/var/www/gsit/node_modules/schema-utils/dist/index.js',
    '/var/www/gsit/node_modules/copy-webpack-plugin/dist/index.js',
    '/var/www/gsit/webpack.config.js',
    '/var/www/gsit/node_modules/webpack-cli/lib/webpack-cli.js',
    '/var/www/gsit/node_modules/webpack-cli/lib/bootstrap.js',
    '/var/www/gsit/node_modules/webpack-cli/bin/cli.js',
    '/var/www/gsit/node_modules/webpack/bin/webpack.js'
  ]
}
```
* Docu points to GLPI. (A simple "Search & Replace" would've sufficed)

-> Using the archive fails, because all dependencies files (composer, npm, webpack) are missing. Adding them results in the same as above.

### Steps To reproduce

Install as described here: https://github.com/DCS-Easyware/gsit/blob/gsit9.6/INSTALL.md

### Anything else?

* Ubuntu 22.04
* PHP 8.1
* NodeJS 20

Looks like errors that happen if you rush a project into a repo.

I basically wanted to try it out. Would've prefered an online demo.
Sadly I guess I have to use the "less secure" GLPI if this one does not install.

**Comment by ddurieux** [comment]
You try to install the branch gsit9.6? or the release archive 9.5.12?

**Comment by blwtxc** [comment]
I tried both

`wget https://github.com/DCS-Easyware/gsit/releases/download/GSIT-9.5.12/gsit-9.5.12.tgz`

and

`git clone https://github.com/DCS-Easyware/gsit.git`

**Comment by blwtxc** [comment]
As I said, using the archive, causes error because all dependencies-files are missing. Adding them results, in the end, in the same errors I get when I clone the repo.

If I clone the repo, I get error because the `.lock` files are present. Removing them resuts in the errors described.

**Comment by ddurieux** [comment]
For the first (9.5.12), you can use php8.1 but have warnings / errors, but you not need patch r install dependancies because you have them inside the archive.

For thr git clone, the actual branch is in development for the 9.6 version, with php 8.1 to 8.3 compatible, new libs... but not yet ready

**Comment by blwtxc** [comment]
Do you see that webpack does not build?

**Comment by ddurieux** [comment]
that is the error you have with `gsit-9.5.12.tgz`?

**Comment by blwtxc** [comment]
Both ... as I said

**Comment by blwtxc** [comment]
Simply uncompressing, copying and pointing php (fpm) and nginx to it, gives me the same output I see the whole day: `File not found`

I'm pretty sure that "everything works" on your side, but that's not my experience ... I'm trying various ways for several hours now.

Basically: For me the documantation does not reflect ALL the steps necessary to make it work.
* No steps do not work
* The steps in INSTALL.md don't
* Going to the readthedocs (of GLPI) doesn't really help either

**Comment by ddurieux** [comment]
File not found on the browser? if yes seems a configuration problem on webserver. Can you share the webserver name you use and the configuration used?

**Comment by blwtxc** [comment]
Nginx

Following is basically the `nginx` config I found in the readthedocs, with some sugar on top.
```
upstream php_fpm_socket {
	server unix:/run/php/php8.1-fpm.sock;
}

server {
	listen 80;
	listen [::]:80;
	
	server_name xxxx.xxx;
	
	# Include location directive for Let's Encrypt ACME Challenge
	include /etc/nginx/snippets/letsencrypt-acme-challenge.conf;
	
	set $DO_REDIR "";
	if ($host = xxxx.xxx) {
		set $DO_REDIR "Y";
	}
	if ($uri !~ "/\.well-known/acme-challenge/") {
		set $DO_REDIR "${DO_REDIR}Y";
	}
	if ($DO_REDIR = YY) {
		return 301 https://$host$request_uri;
	}
	if ($uri !~ "/\.well-known/acme-challenge/") {
		return 404;
	}
}
server {
	listen 443 ssl;
	listen [::]:443 ssl;
	http2 on;
	
	root /var/www/html/public;
	index index.php index.html index.htm index.nginx-debian.html;
	
	server_name xxxx.xxx;
	
	
	error_log /var/log/nginx/xxxx.error;
	access_log /var/log/nginx/xxxx.access;
	
	
	ssl_certificate /etc/letsencrypt/live/xxxx/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/xxxx/privkey.pem; # managed by Certbot
	
	
	add_header Strict-Transport-Security "max-age=31536000";
	
	
	##
	# Error Response Handling
	##
	include /etc/nginx/snippets/error-response.conf;
	
	
	##
	# Client Settings
	##
	client_max_body_size 100M;
	client_body_buffer_size 16k;
	
	
	##
	# Gzip Settings
	##
	gzip_static on;
	gzip on;
	
	
	##
	# Location Settings
	##
	
	## Allow access for .well-known/acme-challenge
	location ~* ^/\.well-known/acme-challenge/ {
		allow all;
		access_log off;
		log_not_found off;
		auth_basic off;
		root /usr/local/acme/;
		try_files $uri $uri/ =404;
	}
	
	# Allow access to '^/.well-known/'
	location ~* ^/\.well-known/ {
		allow all;
		access_log off;
		log_not_found off;
		autoindex off;
	}
	
	# Deny all attempts to access hidden files such as .htaccess.
	location ~ /\. {
		deny all;
	}
	
	# Handling noisy messages
	location ~ ^/favicon(.*)\.(ico|png) {
		access_log off;
		log_not_found off;
	}

	location ~ ^/robots.txt {
		allow all;
		log_not_found off;
		access_log off;
	}
	
	
	##
	# Location Settings
	##

	location / {
		try_files $uri /index.php$is_args$args;
	}

	location ~ ^/index\.php$ {
        fastcgi_split_path_info ^(.+\.php)(/.*)$;
		include fastcgi_params;
		#fastcgi_intercept_errors on;
		fastcgi_pass php_fpm_socket;
		fastcgi_param  SCRIPT_FILENAME $document_root$fastcgi_script_name;
    }
}
```

**Comment by ddurieux** [comment]
replace the location ~ ^/index.... and the lines with fastcgi_split_path (in fact the last lines with

```
        location ~ \.php {
            include fastcgi_params;
            fastcgi_split_path_info ^(.+\.php)(/.+)$;
            fastcgi_pass php_fpm_socket;
            fastcgi_param  SCRIPT_FILENAME  /var/www/html/public/$fastcgi_script_name;
            fastcgi_index index.php;
            fastcgi_param  HTTP_REFERER $http_referer;
            include        fastcgi_params;
            fastcgi_param  SERVER_NAME      $host;
            set $path_info $fastcgi_path_info;
            fastcgi_param  PATH_INFO $path_info;
        }

```
so be sure you have the folders : /var/www/html/public/gsit/ and then inside the file index.php

**Comment by ddurieux** [comment]
And after in the browser https://yourdomain/gsit/

**Comment by blwtxc** [comment]
Why should I do it like that? There's no `gsit` dir in public (there's a `build` and a `lib` dir thou)

**Comment by ddurieux** [comment]
your webserver must use your gsit folder, if it's anywhere else, you must update more in the nginx config file.
If you are on redhat, censot, or any linux distribution with SELINUX, the best is to put in /var/www/ to be sure it works

**Comment by blwtxc** [comment]
Hmmm ... you seem to assume I don't know how to setup my webserver and also that I would send you my config file `as is`.
A question that comes to mind is: "Why should I name the dir of this web application `gsit`?" If, for whatever reason, your application relies on "being in a dir named `gsit`", I would call that a bad design. As you can see, the web-root points into the `public` dir. This is taken from the GLID readthedocs, you link to. If your application expects a different root, then this is not documented.
I don't know what dependencies are present on your server **before** you install your app. It is very likely that its different from mine. I don't believe that it's productive to now post all packages I installed on my machine as it also hosts another 30ish sites I don't intend to break.
So, basically I really whished there was a documentation for *your* app, that would explain all that.

I really think it's very kind of you that you take your time and try to fix my setup, but at this point, I guess I wait for your next push and try again then ... in the meantime I have to find another solution.

**Comment by ddurieux** [comment]
Ok, thanks for information, you're right, we will produce a documentation for the 9.6 release ;)
